daffodil-cli/src/test/scala/org/apache/daffodil/cli/cliTest/TestCLISaveParser.scala

```diff
@@ -318,4 +318,20 @@ class TestCLISaveParser {
     }
   }
 
+  /**
+   * Attempted to save-parser an invalid schema so we can check the diagnostic error for leaked information
+   */
+  @Test def test_CLI_Saving_SaveParser_error(): Unit = {
+    val schema = path(
+      "daffodil-sapi/src/test/resources/test/sapi/mySchema6.dfdl.xsd",
+    )
+
+    withTempFile { parser =>
+      runCLI(args"save-parser -s $schema $parser") { cli =>
+        cli.expectErr("[error]")
+        cli.expectErr(s"Schema context:  Location line 32 column 74 in $schema")
+      }(ExitCode.UnableToCreateProcessor)
+    }
+  }
```

Comment:
I manually ran this in a CLI and the diagostic location looks good, but I got the same error twice:

```
[error] Schema Definition Error: Error loading schema due to src-resolve: Cannot resolve the name 'tns:nonExistent' to a(n) 'type definition' component.
Schema context:  Location line 32 column 74 in daffodil-sapi/src/test/resources/test/sapi/mySchema6.dfdl.xsd
[error] Schema Definition Error: Error loading schema due to src-resolve: Cannot resolve the name 'tns:nonExistent' to a(n) 'type definition' component.
Schema context:  Location line 32 column 74 in daffodil-sapi/src/test/resources/test/sapi/mySchema6.dfdl.xsd
```

I ran the same command on the main branch and only get it once (but with a much less human readable error emssage). I'm not sure what would have changed to break that. But this error does come from xerces and this PR does slightly change how xerces errors are handled, maybe something is wrong there?